### DISCUSSION TITLE
pan and zoom?

### DISCUSSION THREAD

**Discussion by Taubenstrohhalm** [ExperienceLovelace/ha-floorplan#18] [Q&A] [answered]
Hey there,

great work! thank you so much for this add-on!
I'm quite new to home-assistant and therefore to ha-floorplan. Hope I haven't overseen anything in the documentation about a way to pan and zoom in the svg? - did you implement something like this already?

thanks in advance

**Comment by exetico** [comment]
Hi

Are you thinking of a feature like some webshops have, so you could have a very large floorplan, and zoom into a element, with... items from a single room (but with many entities), so it's possible to add both Tv, Speaker, Lights, Aircon and whatever else?

If so, the answer is no - but I like the idéa. However... I can't really see a good way if implement this as a zoom-solution, so maybe you're better with a solution, where the floorplan, just opens another floorplan in a custom card? (With a solution like browsermod)

**Reply by kidcoma** [reply]
I think it would basically be like a GIS, where it queries the objects within the grid (i.e. the extents of your screen) based on the current zoom level. So for each item you would have to specify the zoom level that you would want to start seeing it at. I'm not a big programmer so not sure how the actual zooming works with the UI.

**Reply by Taubenstrohhalm** [reply]
Hmm I like that idea, but would imagine that it's quite complicated to implement and would take quite long to specify the different zoom levels for all components? 
I was more thinking about something like this:
https://observablehq.com/@d3/zoomable-scatterplot
They (d3) are also using svg's. Because its all vectors, I don't think we would have to draw the components in different zoom levels.

**Reply by kidcoma** [reply]
I envision it would just be one parameter per component (start seeing component at zoom level x). Perhaps you could do it via entity domain. I did some Autodesk MapGuide programming about 20 years ago (20 years, yikes!) and if IRCC that's how it worked.

**Comment by exetico** [comment, marked as answer]
I guess it's possible to allow zoom like that, but I'm not really sure that i would like most...  I'm one of many, which hidding a image behind my elements, and in that cause, it's not that useful. But maybe @pkozul can see a good use for it (i can too, i would just not use it myself) :-) 

Well free to create a issue, and do your best to create a good description of it. We can't promise anything, but it's better to have typed down than... no reminder at all :-D
